polonius-engine/src/output/datafrog_opt.rs

```diff
@@ -378,6 +397,47 @@ pub(super) fn compute<T: FactTypes>(
                 &loan_live_at,
                 |&(loan, point), _, _| (loan, point),
             );
+
+            // subset_placeholder(Origin1, Origin2, Point) :-
+            //     subset(Origin1, Origin2, Point),
+            //     placeholder_origin(Origin1).
+            subset_placeholder.from_leapjoin(
+                &subset_o1p,
+                (
+                    placeholder_origin.extend_with(|&((origin1, _point), _origin2)| origin1),
+                    // remove symmetries:
+                    datafrog::ValueFilter::from(|&((origin1, _point), origin2), _| {
+                        origin1 != origin2
+                    }),
+                ),
+                |&((origin1, point), origin2), _| (origin1, origin2, point),
+            );
+
+            // subset_placeholder(Origin1, Origin3, Point) :-
+            //     subset_placeholder(Origin1, Origin2, Point),
+            //     subset(Origin2, Origin3, Point).
```

Comment:
can we add a comment here that we maintain the invariant that `Origin1` is a placeholder region?

Reply:
I've pushed a commit to address this